ethcore/src/client/client.rs

```diff
@@ -439,7 +439,10 @@ impl<V> BlockChainClient for Client<V> where V: Verifier {
 		};
 		// that's just a copy of the state.
 		let mut state = self.state();
-		let sender = try!(t.sender());
+		let sender = try!(t.sender().map_err(|e| {
+			warn!(target: "evm", "Transaction mallformed: {:?}", e);
+			ExecutionError::TransactionMallformed
```

Comment:
Well perhaps from "pattern" should not be used here, because conversion of generic error into concrete error of the function is not one-to-one in general (unlike when you transform in the opposite direction), this is just one case here